alphalens/performance.py

```diff
@@ -329,169 +330,26 @@ def factor_alpha_beta(factor_data,
     return alpha_beta
 
 
-def cumulative_returns(returns, period, freq=None):
```

Comment:
That was my initial thought. I actually refactored the logic for `period` values other than 1D into a function called subportfolio_cumulative_returns, but luca (the original author) convinced me that we should just remove it: https://github.com/quantopian/alphalens/pull/361#discussion_r383133382, and I agree with him. It will just add confusion. We may want to go back and re-add it later, but if it's not getting used in a tearsheet, I don't think it's worth keeping it in.

Reply:
I have just had a last thought on this: why don't we keep the current `cumulative_returns` behaviour and add a special case for the most common scenario that covers 95% of people's use cases? At the beginning of the function we check if we are computing cumulative returns for a daily factor and if so we call `empyrical` otherwise we default to current code. Also we need to plot only 1 day period cumulative returns by default, currently a plot for each period is provided. This would be a less invasive change but still being able to fix the performance issue. On the other hand it would be ugly to compute cumulative returns in two totally different ways depending on the input... I am not sure. What do you think?

Reply:
I do think that would be a somewhat ugly band-aid type fix. Given the state of this PR and prior discussions I'm inclined to keep Jamie's changes for now, and then we can always reinstate the deleted code in a more carefully organized way if we need it back later.

Reply:
Luca, that was my initial thought, but it made the code very hard to follow (from my perspective). I think I agree with David that it would make sense to try to work the logic back in later. Is that ok with you?

Reply:
Sure, no problem. I just wanted to make sure this possibility has been considered.